libraries/Standard/src/math/special.birch

```diff
@@ -581,15 +592,15 @@ function asymptotic_series(a:Real, x:Real, func:Integer) -> Real {
   let flag1 <- true;
   let k <- 1;
   while (k <= K && flag1) {
-    ck <- asymptotic_series_d[k,1];
+    ck <- IGAM_ASYMPTOTIC_SERIES_D[k,1];
     let flag2 <- true;
     let n <- 2;
     while (n <= N && flag2) {
       if (n > maxpow) {
```

Comment:
I might be missing something here, but isn't `maxpow` initialized to `0`, and `n` to `2`, and they each increment by one on each iteration of the loop, so `n > maxpow` is always true? Is that supposed to happen or does it indicate a bug?

Reply:
Good catch, I took a look at the scipy implementation and it appears like this is the case in there as well: https://github.com/scipy/scipy/blob/master/scipy/special/cephes/igam.c#L395

Let me know what you think, if it is, I can open an issue in scipy, find the fix and fix it here too.

Reply:
Ah wait, I've misunderstood; `maxpos` is not reset to zero in each iteration of the outer loop over `k`. So it gets increased on the first iteration of the outer loop, then maintains that higher value for the remaining iterations. So there is something meaningful happening here.